File: backend/blackbox/src/main/java/com/insight_web/blackbox/usecase/action/SiteAction.java
package com.insight_web.blackbox.usecase.action;

import com.insight_web.blackbox.domain.Site;

public class SiteAction {
    private String siteName;
    private String url;

    public Site toDomainModel(Long userId) {
        var site = new Site();
        site.setSiteName(siteName);
        site.setId(url);
        site.setOwnerId(userId);
        return site;
    }
}

File: backend/auth/src/main/kotlin/com/insightweb/auth/usecase/action/SiteAction.kt
package com.insightweb.auth.usecase.action

import com.insightweb.auth.domain.Site

data class SiteAction(
    val siteName: String,
    val url: String,
) {
    fun toSite(userId: Long): Site = Site(
        id = url,
        ownerId = userId,
        siteName = siteName
    )
}
